sdk/src/spectrumx/client.py

```diff
@@ -484,5 +485,87 @@ def upload_capture(
         else:
             return capture
 
+    def _handle_existing_capture_error(
+        self, err: SDSError, captures: list[Capture]
+    ) -> bool:
+        """
+        Handle the case where a capture already exists.
+        Returns True if handled, False otherwise.
+        """
```

Comment:
This needs refactoring: it's a bit confusing as it's returning a bool which meaning is not immediately clear, and also modifying `captures`, which is not a good idea due to the potential of accidentally deleting a pre-existing capture.

Perhaps it make sense to have the `gateway.py` module raise a specific exception for these conflict cases; then have this function return the conflicting capture instead of mutating `captures`.